### PR TITLE
[FIX] web, models: issues in onchange m2m in o2m

### PR DESCRIPTION
**Pull request by Gorash**
[FIX] web, models: issues in onchange m2m in o2m
- allow to use new in onchange to create m2m
- view and widget objects send all sub-field to receive complete server information from onchange (including sub-field for any level)
- add guardrails to find errors and improve the code in the future
- fix the rendering in list editable to display change and m2m tags
- remove some unless change trigger
- add test for x2m (client and server side)

### CODE REVIEW

**Comment by Gorash** [comment]
ping ping @ged-odoo @rco-odoo

File: openerp/addons/test_new_api/tests/test_onchange.py [hunk 1]
```diff
@@ -104,6 +104,7 @@ def test_onchange_one2many(self):
                     'body': BODY,
                     'author': USER.id,
                     'size': len(BODY),
+                    'tags': []
```
Comment (rco-odoo) [inline openerp/addons/test_new_api/tests/test_onchange.py]: please add a trailing comma

File: openerp/addons/test_new_api/tests/test_onchange.py [hunk 2]
```diff
@@ -112,17 +113,20 @@ def test_onchange_one2many(self):
         self.assertIn('messages', result['value'])
         self.assertItemsEqual(result['value']['messages'], [
             (5,),
+            (4, message.id),
             (1, message.id, {
                 'name': "[%s] %s" % ("Foo", USER.name),
                 'body': message.body,
                 'author': message.author.name_get()[0],
                 'size': message.size,
+                'tags': [(5,)]
```
Comment (rco-odoo) [inline openerp/addons/test_new_api/tests/test_onchange.py]: please add a trailing comma

File: openerp/addons/test_new_api/tests/test_onchange.py [hunk 3]
```diff
@@ -112,17 +113,20 @@ def test_onchange_one2many(self):
...
             }),
             (0, 0, {
                 'name': "[%s] %s" % ("Foo", USER.name),
                 'body': BODY,
                 'author': USER.name_get()[0],
                 'size': len(BODY),
+                'tags': [(5,)]
```
Comment (rco-odoo) [inline openerp/addons/test_new_api/tests/test_onchange.py]: please add a trailing comma

**Comment by antonylesuisse** [comment]
And @rco-odoo

**Comment by Gorash** [comment]
@odony everything is good for you?

**Comment by Gorash** [comment]
Branch splitted: 9.0-onchange-subfield-chm and 9.0-new-m2m-o2m-chm

**Pull request closed without merging**